### PR TITLE
Fix Incorrect Use of Reference Equality for INVALID_HANDLE_VALUE

### PR DESCRIPTION
**Pull request by mythili-rajaraman**
Fix for the Issue - https://github.com/oshi/oshi/issues/1765

### CODE REVIEW

**Comment by mythili-rajaraman** [comment]
@dbwiddis , Since I am the first time contributor, you might want to approve the workflow . Thank you!

`First-time contributors need a maintainer to approve running workflows. 1 in progress, 3 pending, and 1 successful checks`

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,8 @@
 # 6.0.0 (in progress)
 
+##### Bug fixes / Improvements
+* [#1768](https://github.com/oshi/oshi/pull/1768): Fix Incorrect use of Reference Equality - [@mythili-rajaraman](https://github.com/mythili-rajaraman).
```
Comment (adrian-kong) [inline CHANGELOG.md:4]: I think if I were Daniel, he would talk about the capitalization and past tense.

Comment (adrian-kong) [reply CHANGELOG.md:4]:
```suggestion
* [#1768](https://github.com/oshi/oshi/pull/1768): Fixed incorrect use of reference equality - [@mythili-rajaraman](https://github.com/mythili-rajaraman).
```

**Review by adrian-kong — APPROVED** [review]

File: oshi-core/src/main/java/oshi/hardware/platform/windows/WindowsPowerSource.java [hunk 2]
```diff
@@ -151,7 +151,7 @@ private static WindowsPowerSource getPowerSource(String name) {
 
         HANDLE hdev = SetupApi.INSTANCE.SetupDiGetClassDevs(GUID_DEVCLASS_BATTERY, null, null,
                 SetupApi.DIGCF_PRESENT | SetupApi.DIGCF_DEVICEINTERFACE);
-        if (WinBase.INVALID_HANDLE_VALUE != hdev) {
+        if (!WinBase.INVALID_HANDLE_VALUE.equals(hdev)) {
```
Comment (adrian-kong) [inline oshi-core/src/main/java/oshi/hardware/platform/windows/WindowsPowerSource.java:154]: Maybe also here, swap the order around so it follows the rest?
```suggestion
        if (!hdev.equals(WinBase.INVALID_HANDLE_VALUE)) {
```

Comment (mprins) [reply oshi-core/src/main/java/oshi/hardware/platform/windows/WindowsPowerSource.java:154]: yes, also if for some reason `hdev` for some unexpected reason turns out to be `null` it prevents a NullPointerException

File: oshi-core/src/main/java/oshi/driver/windows/DeviceTree.java [hunk 3]
```diff
@@ -91,7 +91,7 @@ private DeviceTree() {
         Map<Integer, String> mfgMap = new HashMap<>();
         // Get device IDs for the top level devices
         HANDLE hDevInfo = SA.SetupDiGetClassDevs(guidDevInterface, null, null, DIGCF_DEVICEINTERFACE | DIGCF_PRESENT);
-        if (hDevInfo != INVALID_HANDLE_VALUE) {
+        if (!hDevInfo.equals(INVALID_HANDLE_VALUE)) {
```
Comment (mprins) [inline oshi-core/src/main/java/oshi/driver/windows/DeviceTree.java:94]: better to have 
```suggestion
        if (!INVALID_HANDLE_VALUE.equals(hDevInfo)) {
```

see also below

File: oshi-core/src/main/java/oshi/software/os/windows/WindowsFileSystem.java [hunk 4]
```diff
@@ -181,7 +181,7 @@ public WindowsFileSystem() {
         aVolume = new char[BUFSIZE];
 
         hVol = Kernel32.INSTANCE.FindFirstVolume(aVolume, BUFSIZE);
-        if (hVol == WinBase.INVALID_HANDLE_VALUE) {
+        if (hVol.equals(WinBase.INVALID_HANDLE_VALUE)) {
```
Comment (mprins) [inline oshi-core/src/main/java/oshi/software/os/windows/WindowsFileSystem.java:184]: again better to cal the `equals()` on `INVALID_HANDLE_VALUE`

**Review by mprins — CHANGES_REQUESTED** [review]

File: oshi-core/src/main/java/oshi/hardware/platform/windows/WindowsPowerSource.java (hunk 2 above)
Comment (dbwiddis) [reply oshi-core/src/main/java/oshi/hardware/platform/windows/WindowsPowerSource.java:154]: As @mprins says, it's generally preferred to call `.equals()` from a known non-null constant and that would be preferred here.   One can go read the documentation for [SetupDiGetClassDevs()](https://docs.microsoft.com/en-us/windows/win32/api/setupapi/nf-setupapi-setupdigetclassdevsw) to verify that it doesn't return null so in this case it's fine; but generally better to be defensive!

Comment (dbwiddis) [reply oshi-core/src/main/java/oshi/hardware/platform/windows/WindowsPowerSource.java:154]: (So in this case it's better as is and the others should be switched to match.)

Comment (mythili-rajaraman) [reply oshi-core/src/main/java/oshi/hardware/platform/windows/WindowsPowerSource.java:154]: Thanks for the details. Kept this as in and switched others

File: CHANGELOG.md (hunk 1 above)
Comment (mythili-rajaraman) [reply CHANGELOG.md]: Totally. Done

File: oshi-core/src/main/java/oshi/driver/windows/DeviceTree.java (hunk 3 above)
Comment (mythili-rajaraman) [reply oshi-core/src/main/java/oshi/driver/windows/DeviceTree.java]: Switched it.

File: oshi-core/src/main/java/oshi/software/os/windows/WindowsFileSystem.java (hunk 4 above)
Comment (mythili-rajaraman) [reply oshi-core/src/main/java/oshi/software/os/windows/WindowsFileSystem.java]: Thanks again. Switched it.

**Comment by dbwiddis** [comment]
Looks great! Thanks for contributing to open source!

**Review by dbwiddis — APPROVED** [review]

**Pull request merged**